bitcoin/src/consensus/encode.rs

```diff
@@ -225,7 +225,10 @@ impl<W: Write + ?Sized> WriteExt for W {
     #[inline]
     fn emit_bool(&mut self, v: bool) -> Result<(), io::Error> { self.write_all(&[v as u8]) }
     #[inline]
-    fn emit_slice(&mut self, v: &[u8]) -> Result<(), io::Error> { self.write_all(v) }
+    fn emit_slice(&mut self, v: &[u8]) -> Result<usize, io::Error> {
+        self.write_all(v)?;
+        Ok(v.len())
```

Comment:
Why? This looks completely redundant since the caller can just call `.len()` on the argument. You've complained about breaking changes and this is a breaking change...

Reply:
>Why? This looks completely redundant since the caller can just call .len() on the argument.

I think it's worthwhile to have a uniform API. You can see at the call sites that it makes the code easier to read.

>You've complained about breaking changes and this is a breaking change...

I'd prefer we take the opposite tack and say "we let you make a breaking change so please don't complain about our breaking changes" ;).

More seriously, while this is a breaking change, it's in an obscure method of a trait that basically nobody ever uses directly. So I think the breakage is worth the API cleanup.

Reply:
> "we let you make a breaking change so please don't complain about our breaking changes"

:laughing: OK.